tests/testthat/test-init.R

```diff
@@ -77,3 +77,20 @@ testthat::test_that("init throws when dataname in filter incompatible w/ datanam
     "Filter 'iris Species' refers to dataname not available in 'data'"
   )
 })
+
+testthat::test_that("init creates a lockfile during the execution", {
```

Comment:
test for failed lockfile creation would be nice also

Reply:
We use `renv::snapshot` with `force = TRUE` - this forces the generation of a lockfile even when pre-flight validation checks have failed. So the file will be created anyway.

It's not like that the file gets created or not. The file gets created even though it might have data that can only be partially restorable. So for the notifications you mentioned in here https://github.com/insightsengineering/teal/pull/1255#discussion_r1672837226

Instead of `Lockfile failed to create.` we put an info that `lockfile created with issues.`

However, we can try to develop a test that returns this lockfile that was created with issues

Reply:
But it's hard to know if the lockfile has issues in it, once it's created. I think the only way is the attempt to `renv::restore()` that fails.